### DISCUSSION TITLE
How do I stop losing my libraries and tags?

### DISCUSSION THREAD

**Discussion by CentaurWorks** [TagStudioDev/TagStudio#367] [Q&A]
Just had this happen a second time, THANK GOODNESS that the backup stave helped

To reload the library with the backup:
Close your gallery
Go to the folder for the gallery
Copy the most recent file in "Backups"
Paste and have it match the library file before deleting the old file
Reopen your gallery

Sadly, I had to learn that the hardway. Is there any way this can be prevented? If not, is there any possibility you can get logs from this software to help with future developments?

**Comment by Bamowen** [comment]
Hey, if still relevant, have you opened a Bug report for this? With the system info as well? :)

**Reply by Tishj** [reply]
I think this is an issue with 9.4  I can confirm this has happened to me with 9.4, with 9.5 this hasn't happened yet. Another bonus of using sqlite as a backend

**Reply by CyanVoxel** [reply]
9.4.2 introduced an automatic backup feature (#554) to address the issue in #343 which seems to be the same issue described here. Though now that 9.5 uses the SQLite save file, this *shouldn't* be an issue with releases moving forward
